examples/karpenter/main.tf

```diff
@@ -31,6 +31,11 @@ provider "helm" {
       args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
     }
   }
+  registry {
+    url      = "oci://public.ecr.aws/"
+    username = data.aws_ecrpublic_authorization_token.token.user_name
```

Comment:
whether users want to manage the secrets via the provider or the resource is left up to them. we have shown in the resource because more than likely they are deploying other resources which will now fail using these ECR credentials

Reply:
OK, if you want to proceed with this PR, I can remove this. In my case, I manage my helm installations with other tooling and found that the extra 5-10 second wait to upgrade the helm chart for an updated token was needless while I iterated on karpenter options. I really haven't used terraform long enough to know what the ramifications are of using a provider vs resource. Thank you for the feedback.